testing/dart/compile_test.gni

```diff
@@ -0,0 +1,61 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import("//third_party/dart/build/dart/dart_action.gni")
+
+# Generates a Dart kernel snapshot using flutter_frontend_server.
+#
+# Arguments
+#     dart_main (required): The Main Dart file.
+#
+#     dart_kernel (required): The path to the output kernel snapshot in the out
+#                             directory.
+#
+#     packages (required): The path to the .packages file.
+template("compile_flutter_dart_test") {
+  assert(defined(invoker.dart_file), "The Dart test file must be specified.")
+  assert(defined(invoker.dart_kernel),
+         "The Dart Kernel file location must be specified.")
+  assert(defined(invoker.packages),
+         "The path to the .packages file must be specified.")
+
+  dart_action(target_name) {
```

Comment:
I'm pretty sure this will only work if the prebuilt Dart is using the same kernel version as the Dart we're building.

Instead you can use the vm package to directly invoke the kernel compiler from third_party/dart

Reply:
`dart_action` uses the built Dart VM. `prebuilt_dart_action` uses the Dart VM from `//third_party/dart/tools/sdks/dart-sdk/bin/dart`

Reply:
ahhhh ok